app/ts/background/background.ts

```diff
@@ -50,13 +55,32 @@ async function updateMetadataForSimulation(simulationState: SimulationState, eth
 	}
 }
 
+const transactionSpecificVisualisations = async (simulatedTransaction: SimulatedTransaction, ethereum: EthereumClientService) => {
+	try {
+		// identifies compound governane call and performs simulation if the vote passes 
+		if (simulatedTransaction.multicallResponse.statusCode !== 'success') return
+		const fourByte = get4Byte(simulatedTransaction.signedTransaction.input)
+		if (fourByte === undefined) return
+		const explanation = FourByteExplanations[fourByte]
+		if ((explanation !== 'Submit vote' && explanation !== 'Cast vote') || simulatedTransaction.multicallResponse.events.length !== 1) return
+		if (simulatedTransaction.signedTransaction.to === null) return
+		const governanceContractInterface = new Interface(CompoundGovernanceAbi)
+		const params = governanceContractInterface.decodeFunctionData(explanation === 'Submit vote' ? 'submitVote' : 'castVote', dataStringWith0xStart(simulatedTransaction.signedTransaction.input))
+		const contractExecutionResult = await simulateCompoundGovernanceExecution(ethereum, simulatedTransaction.signedTransaction.to, params[0])
+		console.log(contractExecutionResult)
```

Comment:
looks like some debugging code left here, also this silently fails?

Reply:
> looks like some debugging code left here, also this silently fails?

its not debugging code, its the output of the thing, thats all the function does, prints the results to console. Yeah, it should fail silently for some cases, but not others,I'll add issue about it